src/amp-story-player/amp-story-player-impl.js

```diff
@@ -241,14 +297,37 @@ export class AmpStoryPlayer {
   /** @private */
   initializeIframes_() {
     for (let idx = 0; idx < MAX_IFRAMES && idx < this.stories_.length; idx++) {
-      const story = this.stories_[idx];
-      this.buildIframe_(story);
+      this.createIframeForStory_(idx);
+    }
+  }
+
+  /**
+   * Creates an iframe for a certain story. Should only be done if
+   * this.iframes_.length < this.MAX_IFRAMES. It is assumed that iframes
+   * are created for stories in order, starting from the first one.
```

Comment:
> It is assumed that iframes are created for stories in order, starting from the first one.

Is there any case where this would not apply? using any of the other callbacks?

Reply:
As this method is used currently (while setting up the iframes for the first time on layout, or when adding), it is always the case that the stories are created iframes for in order.

As long as this is always used when there are less iframes than the max (and not skipped for some reason), it should mean that there are as many stories as there are iframes. Hence, any index for a story should be the same for its iframe.

Reply:
Originally the method used `find` for the iframe index, but because it would end up finding the same number as the story index, didn't include it. Would it be better if I brought that back?